lib/renaissance/users/user.ex

```diff
@@ -0,0 +1,32 @@
+defmodule Renaissance.User do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  schema "users" do
+    field :email, :string, unique: true
+    field :password, :string, virtual: true
+    field :password_hash, :string
+
+    timestamps(type: :utc_datetime)
+  end
+
+  @doc false
+  def changeset(user, attrs) do
+    user
+    |> cast(attrs, [:email, :password])
+    |> validate_required([:email, :password])
+    |> unique_constraint(:email)
+    |> hash_password()
+  end
+
+  defp hash_password(attrs) do
```

Comment:
Couldn't you deconstruct password out of attrs in `hash_password(attrs)` like pattern match the nil out?

Reply:
Thanks for the feedback, we've talked about the pattern and we're looking into this! 😄